GeoChat.Geolocation/Controllers/ServerController.cs

```diff
@@ -0,0 +1,53 @@
+﻿using GeoChat.Geolocation.Api.DbAccess;
+using GeoChat.Geolocation.Api.Entities;
+using GeoChat.Geolocation.Api.Repo;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.EntityFrameworkCore;
+
+namespace GeoChat.Geolocation.Api.Controllers;
+
+[ApiController]
+[Route("api/[controller]")]
+public class ServerController : ControllerBase
+{
+	private readonly IGenericRepo<Server> serversRepository;
+	private readonly IGenericRepo<Location> locationsRepository;
+
+	[HttpGet("{latitude}/{longitude}")]
+	[Authorize]
+	public async Task<Server> GetServerUrl(double latitude, double longitude)
+	{
+		if (latitude < -90 || latitude > 90) throw new ArgumentException("Wrong latitude value!");
+		if (longitude < -180 || latitude > 180) throw new ArgumentException("Wrong longitude value!");
+		IGeoHasher geoHasher = new GeoHasher();
```

Comment:
Use dependency injection; u can add the dependency in Program.cs using builder.Services.AddScoped<IGeoHasher, GeoHasher>();